pytorch_lightning/accelerators/cpu_backend.py

```diff
@@ -32,10 +32,12 @@ def setup(self, model):
 
         # CHOOSE OPTIMIZER
         # allow for lr schedulers as well
-        optimizers, lr_schedulers, optimizer_frequencies = self.trainer.init_optimizers(model)
-        self.trainer.optimizers = optimizers
-        self.trainer.lr_schedulers = lr_schedulers
-        self.trainer.optimizer_frequencies = optimizer_frequencies
+        if not self.trainer.testing:
+            optimizers, lr_schedulers, optimizer_frequencies = self.trainer.init_optimizers(model)
+            self.trainer.optimizers = optimizers
+            self.trainer.lr_schedulers = lr_schedulers
+            self.trainer.optimizer_frequencies = optimizer_frequencies
```

Comment:
this is a repetitive code in every accelerator. Should be moved to the `Accelerator` class?

Reply:
this will be cleaned later, right? @williamFalcon

Reply:
for now, I moved it to the Accelerator.